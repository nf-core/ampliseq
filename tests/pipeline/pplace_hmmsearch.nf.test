nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_pplace_hmmsearch"
    tag "pipeline"

    test("test_pplace_hmmsearch") {

        when {
            params {
                outdir     = "$outputDir"
                skip_qiime = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/cutadapt/cutadapt_summary.tsv")).match("cutadapt") },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert snapshot(path("$outputDir/input/Samplesheet.tsv"),
                                path("$outputDir/input/Metadata.tsv")).match("input") },
                { assert new File("$outputDir/pplace/bac16s.graft.bac16s.epa_result.newick").exists() },
                { assert new File("$outputDir/pplace/bac16s.taxonomy.per_query_unique.tsv").exists() },
                { assert new File("$outputDir/pplace/bac16s.heattree.tree.svg").exists() },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt")).match("multiqc") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() }
            )
        }
    }
}
