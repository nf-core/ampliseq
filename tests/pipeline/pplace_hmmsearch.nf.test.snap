{
    "input": {
        "content": [
            "Samplesheet.tsv:md5,dbf8d1a2b7933dab9e5a139f33c2b1f4",
            "Metadata.tsv:md5,060b56528bb566eed71f6dfdb52cc395"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.606845"
    },
    "cutadapt": {
        "content": [
            "cutadapt_summary.tsv:md5,5d02749984a811479e7d534fda75163f"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.536473"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,97d0949d7b4bf23f44788530dd6728ad"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.406044"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,864c3e0dc9b4a7649beee0c8665dceb5",
            "ASV_table.tsv:md5,2618251e597593e4d716dd9bed095539",
            "DADA2_stats.tsv:md5,54a1ac8d6c5a3ff15f700c4b2dd40c86",
            "DADA2_table.rds:md5,d095501019ce7ebccfa0eb801db1ed29",
            "DADA2_table.tsv:md5,5c9fb0bfd70da165f0ce6a361bfe0b43"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.570997"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6dae470aace9293d5eb8c318584852dd",
            "rrna.bac.gff:md5,439a9084f089120f700f938dfb58fa41",
            "rrna.euk.gff:md5,c9bc1d9d8fb77dc19c95dee2d53840eb",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.498844"
    },
    "multiqc": {
        "content": [
            "multiqc_general_stats.txt:md5,7a395be7984aaa0f9154de4abed5d824",
            "multiqc_cutadapt.txt:md5,b4409890ffb9c6938433c374b50c380e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-21T15:49:18.662649"
    }
}